lib/gather-fragment-imports.js

```diff
@@ -36,13 +36,24 @@ function gatherFragmentImports({
    * @type {Map<number, Map<string, object>>}
    */
   const lineToFragmentDefinitions = new Map();
-  const importLinesToInlinedSource = inlineImports.lineToImports(source, {
-    resolveOptions: {
-      basedir: path.dirname(sourceLocation),
-    },
-    resolveImport,
-    throwIfImportNotFound,
-  });
+  let importLinesToInlinedSource;
+  if (source?.length > 0 && sourceLocation !== undefined) {
+    importLinesToInlinedSource = inlineImports.lineToImportsFromFile(sourceLocation, {
+      resolveOptions: {
+        basedir: path.dirname(sourceLocation),
+      },
+      resolveImport,
+      throwIfImportNotFound,
+    });
+  } else {
+    importLinesToInlinedSource = inlineImports.lineToImports(source, {
+      resolveOptions: {
+        basedir: path.dirname(sourceLocation),
+      },
+      resolveImport,
+      throwIfImportNotFound,
+    });
+  }
```

Comment:
we should not be doing `path.dirname(undefined`)

`sourceLocation` is a required parameter - let's add a check and throw if it's `undefined` rather than do this branching